app/models/order_detail.rb

```diff
@@ -856,9 +855,17 @@ def has_completed_reservation?
     !product.is_a?(Instrument) || (reservation && (reservation.canceled_at || reservation.actual_end_at || reservation.reserve_end_at < Time.zone.now))
   end
 
+  def time_for_policy_lookup
+    if fulfilled_at
+      fulfilled_at
+    elsif reservation.try(:canceled?)
+      Time.current
```

Comment:
Could this method just be `fulfilled_at || Time.current`?
I'm wondering since `Product#cheapest_price_policy` should default to `Time.zone.now` if given a falsy `date`.